tsdb/index/tsi1/index.go

```diff
@@ -1,131 +1,145 @@
 package tsi1
 
 import (
-	"crypto/rand"
-	"encoding/json"
 	"errors"
 	"fmt"
-	"io/ioutil"
 	"os"
 	"path/filepath"
 	"regexp"
+	"runtime"
 	"sort"
 	"strconv"
-	"strings"
 	"sync"
-	"time"
+	"sync/atomic"
 
+	"github.com/cespare/xxhash"
 	"github.com/influxdata/influxdb/models"
-	"github.com/influxdata/influxdb/pkg/bytesutil"
 	"github.com/influxdata/influxdb/pkg/estimator"
+	"github.com/influxdata/influxdb/pkg/estimator/hll"
+	"github.com/influxdata/influxdb/pkg/slices"
 	"github.com/influxdata/influxdb/query"
 	"github.com/influxdata/influxdb/tsdb"
 	"github.com/influxdata/influxql"
 	"go.uber.org/zap"
 )
 
-const (
-	// IndexName is the name of the index.
-	IndexName = "tsi1"
-
-	// Version is the current version of the TSI index.
-	Version = 1
-)
-
-// Default compaction thresholds.
-const (
-	DefaultMaxLogFileSize = 5 * 1024 * 1024
-)
+// IndexName is the name of the index.
+const IndexName = "tsi1"
 
 func init() {
-	tsdb.RegisterIndex(IndexName, func(id uint64, database, path string, opt tsdb.EngineOptions) tsdb.Index {
-		idx := NewIndex()
-		idx.ShardID = id
-		idx.Database = database
-		idx.Path = path
-		idx.options = opt
+	// FIXME(edd): Remove this.
+	if os.Getenv("TSI_PARTITIONS") != "" {
```

Comment:
I'm OK merging with this, but we need to add a ticket and come up with a strategy for either making the number of partitions configurable, or empirically determining an appropriate way to set it.

Reply:
Issue to track this: https://github.com/influxdata/influxdb/issues/9156